src/agentscope/rpc/rpc_meta.py

```diff
@@ -0,0 +1,218 @@
+# -*- coding: utf-8 -*-
+""" Meta class for all classes that can run on rpc server."""
+from abc import ABCMeta
+from typing import Any, Callable
+import uuid
+from loguru import logger
+
+from .rpc_object import RpcObject, _ClassInfo
+from .retry_strategy import RetryBase, _DEAFULT_RETRY_STRATEGY
+
+
+# Decorator for async and sync functions
+
+
+def async_func(func: Callable) -> Callable:
+    """A decorator for async function.
+
+    In distributed mode, async functions will return a `AsyncResult`
+    immediately.
+
+    Args:
+        func (`Callable`): The function to decorate.
+    """
+
+    func._is_async = True  # pylint: disable=W0212
+    return func
+
+
+def sync_func(func: Callable) -> Callable:
+    """A decorator for sync function.
+
+    In distributed mode, sync functions will block the current thread until
+    the result is ready.
+
+    In most cases, you don't need to use this decorator. `RpcMeta` will
+    treat all public functions without `async_func` as `sync_func`.
+    However, for magic methods (e.g. `__str__` and `__getitem__`, which are
+    started with `__`), you can use `sync_func` to mark them as sync.
+
+    Args:
+        func (`Callable`): The function to decorate.
+    """
+    func._is_sync = True  # pylint: disable=W0212
+    return func
+
+
+# TODO: add stream function decorator `stream_func`
+
+
+def generate_oid() -> str:
```

Comment:
A minor question: why is it named `oid`?

Reply:
short term of `object_id`